src/main/java/org/apache/sling/commons/scheduler/impl/QuartzThreadPool.java

```diff
@@ -77,14 +97,28 @@ public boolean runInThread(final Runnable job) {
      */
     @Override
     public int blockForAvailableThreads() {
-        return this.executor.getConfiguration().getMaxPoolSize() - this.executor.getConfiguration().getQueueSize();
+        synchronized ( this.lock ) {
+            while ( this.counter == 0 ) {
+                try {
+                    this.lock.wait();
+                } catch (final InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                }
```

Comment:
another paranoia idea :
```
            while ( this.counter <= 0 ) {
                try {
                    this.lock.wait(1000);
                } catch (final InterruptedException e) {
                    Thread.currentThread().interrupt();
                }
```

Reply:
Same as above for the <= check :) For the timeout, this might unnecessarily burn some cpu - which is in most case not necessary